### DISCUSSION TITLE
[v4] Multi theme font sans not works for data theme

### DISCUSSION THREAD

**Discussion by rpiechowicz** [tailwindlabs/tailwindcss#15434] [Help] [answered]
**What version of Tailwind CSS are you using?**

v4-0.0.8

**What build tool (or framework if it abstracts the build tool) are you using?**

--

**What version of Node.js are you using?**

--

**What browser are you using?**

Chrome

**What operating system are you using?**

macOS

**Reproduction URL**

https://play.tailwindcss.com/lJzDReT0aS?size=540x720&file=css

**Describe your issue**

Fails to overwrite font sans for custom data theme.
<img width="408" alt="image" src="https://github.com/user-attachments/assets/39ec3818-b2bb-43a9-839c-e65465dc56f8" />

Reproduction bug in tailwind play.

**Comment by adamwathan** [comment, marked as answer]
Hey! Here's a working version of what you're trying to do:

https://play.tailwindcss.com/pFXjFPyhb1?size=540x720&file=css

Key points:

- Should use `@layer theme` (although `@layer base` should work too, but that's a bug you helped me just notice in Tailwind Play)
- Need to make sure you set `font-sans` on the same element that has the data attribute, the font won't update otherwise because it's inherited from higher in the tree and the place where it inherits from resolves the CSS variable to a different value (this is just CSS behavior, not Tailwind stuff)

**Reply by Arthurjim** [reply]
Hi, is this not working with a prefix?

![image](https://github.com/user-attachments/assets/08bee442-97ae-4126-96e2-3d120881dafd)

**Reply by wongjn** [reply]
You'd need to prefix your CSS variables with your prefix too: https://play.tailwindcss.com/RsE8DSqGgK?file=css
